cgrok/tree.c

```diff
@@ -0,0 +1,197 @@
+#include "tree.h"
```

Comment:
The shim that allows libdict to replace tokyocabinet. I wrote this from scratch, focus here :)